### PR TITLE
Define and use augmented move_base goal

### PR DESCRIPTION
**Pull request by jasonimercer**
Part of a set of 5 PRs to enable arbitrary data in a move_base goal. 

@skaynama @p6chen @abencz

### CODE REVIEW

**Comment by skaynama** [comment]
LGTM. One minor offline comment about the attribute_helper.

File: move_base/src/move_base.cpp [hunk 1]
```diff
@@ -826,20 +829,35 @@ namespace move_base {
...
       goal_manager_->setActiveGoal(false);  // setting no active goal
       return;
     }
 
-    geometry_msgs::PoseStamped goal = goalToGlobalFrame(move_base_goal->target_pose);
+    geometry_msgs::PoseStamped goal = goalToGlobalFrame(move_base_goal->goal.pose_stamped);
     goal_manager_->setCurrentGoal(nav_core::NavGoal(goal));
 
+    // Get the XY goal tolerance with a default of the param file value
+    autonomy_msgs::getAttribute(move_base_goal->goal,
```
Comment (p6chen) [inline move_base/src/move_base.cpp]: Should this be separate tolerances for X and Y?

Comment (p6chen) [inline move_base/src/move_base.cpp]: Actually nvm, I see this is probably just replicating the current tolerance system.

**Comment by p6chen** [comment]
lgtm

File: move_base/src/move_base.cpp (hunk 1 above)
Comment (jasonimercer) [inline move_base/src/move_base.cpp]: You are correct. Just replicating for now, we can decide how to change or add attributes next.

**Comment by abencz** [comment]
Instead of modifying the PoseStamped message, why not include the attributes alongside a normal PoseStamped?

**Comment by jasonimercer** [comment]
Didn't know multiple types could define an action. Will remove wrapper type.

**Comment by abencz** [comment]
I think you can remove the dependency on autonomy_msgs in move_base. It now depends on autonomy_msgs only implicitly through move_base_msgs.

**Comment by abencz** [comment]
Not a bad idea to squash these commits as well.

**Comment by jasonimercer** [comment]
I'll remove the dep and allow it to come in via move_base_msgs. I'll squash the commits for all the PRs before I merge.

**Pull request merged**